### ISSUE TITLE
`nesting` rule no longer catching nested code

### ISSUE THREAD

**Issue by stepheng** [realm/SwiftLint#3496]
### New Issue Checklist

- [ X ] Updated SwiftLint to the latest version
- [ X ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Existing `nesting` rule behaviour has changed so that existing violations are no longer triggered. Previously was on 0.40.3 and issue appeared after updating to 0.42.

Can nest classes two deep and can nest functions three deep (default is two) without triggering the error.

```swift
class Nesting {
    class Nested {
        class NestedAgain { } // triggers violation
    }
}
```

Overriding the defaults, setting `type_level` to 0 would trigger on `class Nested {` with the warning `Types should be nested at most 0 level deep (nesting)` which is misleading. It also catches on `typealias`:

```swift
class Nesting {
    typealias NestedTypeAlias = String // Nesting Violation: Types should be nested at most 0 level deep (nesting)
}
```

```swift
class Nesting {
    func function() {
        func nestedFunction() {
            func anotherNestedFunction() {
                func oneMoreNestedFunction() { } // triggers violation
            }
        }
    }
}
```

Overriding the defaults, setting `function_level` to 0 would trigger on `nestedFunction()` with the warning `Functions should be nested at most 0 levels deep (nesting)` which is misleading.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint --path NestedCode.swift

Linting Swift files at paths NestedCode.swift
Linting 'NestedCode.swift' (1/1)
.../NestedCode.swift:21:1: warning: Line Length Violation: Line should be 120 characters or less: currently 131 characters (line_length)
.../NestedCode.swift:12:9: warning: Nesting Violation: Types should be nested at most 1 level deep (nesting)
.../NestedCode.swift:12:9: warning: Nesting Violation: Types should be nested at most 1 level deep (nesting)
.../NestedCode.swift:21:17: warning: Nesting Violation: Functions should be nested at most 2 levels deep (nesting)
.../NestedCode.swift:25:13: warning: Nesting Violation: Types should be nested at most 1 level deep (nesting)
Done linting! Found 5 violations, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.42
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Package install and Portable
* Paste your configuration file:

Issue is reproducible without configuration file. Tested with:

```yaml
nesting:
  type_level: 1
  function_level: 2
  check_nesting_in_closures_and_statements: true
  always_allow_one_type_in_functions: false
```

Which matches the documented defaults and reproduces the same behaviour as with no configuration file.

Also tried:

```yaml
nesting:
  type_level: 0
  function_level: 0
  check_nesting_in_closures_and_statements: true
  always_allow_one_type_in_functions: false
```

Which alters the behaviour as described in the documentation but is not an acceptable workaround for the issue.

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 12.3
* Do you have a sample that shows the issue?

```swift
class Nesting {
    class Nested { // Expected violation to be triggered by this
        class NestedAgain { // Nesting Violation: Types should be nested at most 1 level deep (nesting)
            func bar() {}
        }
        func foo() {}
    }

    func function() {
        func nestedFunction() { // No violation triggered by this
            func anotherNestedFunction() { // Or, by this
                func yetAnotherNestedFunction() {} // Nesting Violation: Functions should be nested at most 2 levels deep (nesting)
            }
        }
        class FunctionNested { // No violation to be triggered by this
            class NestedAgain { // Nesting Violation: Types should be nested at most 1 level deep (nesting)
            }
            func foo() {} // No violation
        }
    }
}
```
